plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProvider.java

```diff
@@ -607,20 +687,61 @@ public void clearProviderCache() {
     this.roots = new HashMap<>();
   }
 
-  @Override public VFSFile createDirectory( String parentPath, VFSFile file, String newDirectoryName ) {
+  @Override public VFSFile createDirectory( String parentPath, VFSFile fileParent, String newDirectoryName ) {
     try {
-      FileObject fileObject = KettleVFS
-        .getFileObject( file.getPath() + VFSFile.DELIMITER + newDirectoryName, new Variables(),
-          VFSHelper.getOpts( bowl, file.getPath(), file.getConnection(), new Variables() ) );
+      String newDirectoryPath = fileParent.getPath() + VFSFile.DELIMITER + newDirectoryName;
+      FileObject fileObject = getFileObject( newDirectoryPath, new Variables() );
       fileObject.createFolder();
-
-      return VFSDirectory.create( parentPath, fileObject, file.getConnection(), file.getDomain() );
-    } catch ( KettleFileException | FileSystemException | MetaStoreException ignored ) {
+      return createVFSDirectory( parentPath, fileObject );
+    } catch ( FileException | FileSystemException ignored ) {
       // Ignored
     }
     return null;
   }
 
+  /**
+   * Wrapper around {@link KettleVFSService#getFileObject(String, VariableSpace)}
+   * @param file
+   * @param space
+   * @return
+   * @throws FileException
+   */
```

Comment:
Wrapper functions as to follow DRY principle (Don't Repeat Yourself). It makes updates easier in the future.